readthedocs/projects/views/public.py

```diff
@@ -117,23 +117,30 @@ def project_badge(request, project_slug):
         style = "flat"
     badge_path = 'projects/badges/%s-' + style + '.svg'
     version_slug = request.GET.get('version', LATEST)
+    version = None
+    url = badge_path % 'unknown'
+
     try:
         version = Version.objects.public(request.user).get(
             project__slug=project_slug, slug=version_slug)
     except Version.DoesNotExist:
-        url = static(badge_path % 'unknown')
-        return HttpResponseRedirect(url)
-    version_builds = version.builds.filter(type='html',
-                                           state='finished').order_by('-date')
-    if not version_builds.exists():
-        url = static(badge_path % 'unknown')
-        return HttpResponseRedirect(url)
-    last_build = version_builds[0]
-    if last_build.success:
-        url = static(badge_path % 'passing')
-    else:
-        url = static(badge_path % 'failing')
-    return HttpResponseRedirect(url)
+        pass
+
+    if version:
+        version_builds = version.builds.filter(type='html',
+                                               state='finished').order_by('-date')
+        if version_builds.exists():
+            last_build = version_builds[0]
+            if last_build.success:
+                url = badge_path % 'passing'
+            else:
+                url = badge_path % 'failing'
+
+    with staticfiles_storage.open(url) as fd:
+        return HttpResponse(
+            fd.read(),
+            content_type='image/svg+xml',
```

Comment:
Isn't here the place where we want to add some specific cache headers for lowering down the cache time?

Reply:
The method itself has the `@never_cache` decorator which handles that.

Reply:
Perfect! I'm sorry, I didn't see it.